OneCommunityGlobal/HighestGoodNetworkApp
# Ilya log tool frontend

<img width="614" alt="tools-log-task" src="https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/105448137/3b8a15f7-2184-43d9-85bc-39274939e55e">


# Description
Adds functionality for the construction manager to navigate to [bmdashboard/tools/log](http://localhost:3000/bmdashboard/tools/log), select a project, select Check In or Check Out and view a list of all tool type available for the selected action (example: Power Drill, Jackhammer, etc). The manager can then select the items of that type they want to check in or our based on their individual human-readable code, and click "submit". Checking out a tool item will make it unavailable for another check out until it is checked back in. The action will also update the "available" and "using" fields on each tool type and logRecord field on each item in the database.  

## Related PRS (if any):
This frontend PR is related to the [#964](https://github.com/OneCommunityGlobal/HGNRest/pull/964) backend PR.
To test this backend PR you need to checkout the #XXX frontend PR.
…

## Main changes explained:
- New route created for **bmdashboard/tools/log**
- Tools log form created with a multi-select populated with human-readable codes (LogTools.jsx)
- New Redux actions to make a POST request, handle results and errors (invTypeActions.js)
- Each result and/or error are displayed with a "Toast" component 
- The form clears on "Submit" or "Cancel"

## How to test:
1. Check out **ilya_log_tool_backend** on the back end. Make sure you are using Node version 14.17.6. Run **npm install** and **npm run build && npm start** to run the backend locally
2. Check out the current branch on the frontend. Run commands **npm install** and **npm run start:local** to launch the front end application
3. Log in as any user
4. Go to [/bmdashboard](http://localhost:3000/bmdashboard) and log in again
5. Navigate to [/bmdashboard/tools/log](http://localhost:3000/bmdashboard/tools/log).
6. Select a project from the dropdown, select an action (default is "Check In")
7. On the table under "Tool/Equipment Number" select a tool item or multiple items based on the individual code 
8. Press "Submit". A green toast message should display for every tool item selected
9. The table should rerender and the codes selected earlier should no longer be available for the same action
10. OPTIONAL: to verify the changes in the database, open Compass or Robo3t, connect to hgnData_dev, select ** buildingInventoryTypes** collection, enter this query: {"__t":"tool_type","name":"Power Drill"} (change name based on the tool type you selected for check in/out) and press "Find". 
11. Fields "using" and "available" should have more or fewer items after the form was submitted, depending on the action. "Check In" should take an item out of the "using" array and add it to "available". "Check Out" should take and item out of the "available" array and add it to "using".  

## Screenshots or videos of changes:

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/105448137/f2ca5f09-8ade-418d-8318-87218b1a728a
